### DISCUSSION TITLE
What's the use of var _ Querier = (*Queries)(nil)

### DISCUSSION THREAD

**Discussion by FotiadisM** [sqlc-dev/sqlc#2055] [Q&A] [answered]
I don't understand what this line does or why is it useful. Could some pleas explain it me? Thank you very much

```go
type Querier interface {
    // ...
}

var _ Querier = (*Queries)(nil)
```

**Comment by go-aegian** [comment, marked as answer]
var _ Querier = (*Querier)(nil)

allows compiler to verify a struct implements the interface
